scripts/testCardTypes.js

```diff
@@ -1,2 +1,137 @@
-// @todo: Put valueable logic here
-console.log('This file should check files structure of card-types.');
+/* eslint-disable global-require,import/no-dynamic-require */
+const { lstatSync, readdirSync, writeFileSync } = require('fs');
+const { join } = require('path');
+const { tmpdir } = require('os');
+const AnkiExport = require('anki-apkg-export').default;
+
+const OUTPUT_DIR = tmpdir();
+
+require('json5/lib/register');
+
+const isDirectory = source => lstatSync(source).isDirectory();
+const getDirectories = source =>
+  readdirSync(source)
+    .map(name => join(source, name))
+    .filter(isDirectory);
+
+const currentTypes = getDirectories('./types');
+
+const testErrors = {};
+
+const containsFile = fileName => typeDir => {
+  let ret = true;
+  try {
+    lstatSync(`./${typeDir}/${fileName}`);
+  } catch (e) {
+    console.log(e.message);
+    ret = false;
+  }
+  return ret;
+};
+
+const containsReadme = containsFile('README.md');
+const containsExample = containsFile('example.json5');
+const containsValidator = containsFile('validator.js');
+const containsIndex = containsFile('index.js');
+
+const testValidator = typeDir => {
+  let ret = true;
```

Comment:
Could you rename this variable like `result`, `isValid` or similar, please.

Reply:
ok